spec/checkMergeConflictSpec.js

```diff
@@ -0,0 +1,261 @@
+      id: 1,
+      cert: 'test',
+      githubToken: 'test',
+    });
+
+    app = robot.load(oppiaBot);
+    spyOn(app, 'auth').and.resolveTo(github);
+    spyOn(
+      checkMergeConflictModule,
+      'checkMergeConflictsInAllPullRequests'
+    ).and.callThrough();
+    spyOn(
+      checkMergeConflictModule,
+      'checkMergeConflictsInPullRequest'
+    ).and.callThrough();
+  });
+
+  describe('pull request merge results in merge conflict', () => {
+    beforeEach(async () => {
+      // Simulate a merged pull request.
+      payloadData.payload.pull_request.merged = true;
+      payloadData.payload.action = 'closed';
+
+      const pullRequest = { ...payloadData.payload.pull_request };
+      pullRequest.merged = false;
+      // Simulate merge conflict in PR.
+      pullRequest.mergeable = false;
+
+      github.pulls = {
+        get: jasmine.createSpy('get').and.resolveTo({
+          data: pullRequest,
+        }),
+        list: jasmine.createSpy('list').and.resolveTo({
+          data: [pullRequest],
+        }),
+      };
+
+      await robot.receive(payloadData);
+    });
+
+    it('should check for merge conflict', () => {
+      expect(
+        checkMergeConflictModule.checkMergeConflictsInAllPullRequests
+      ).toHaveBeenCalled();
+      expect(
+        checkMergeConflictModule.checkMergeConflictsInPullRequest
+      ).toHaveBeenCalled();
+    });
+
+    it('ping pr author regarding merge conflict', () => {
+      expect(github.issues.createComment).toHaveBeenCalled();
+
+      const link = 'link'.link(
+        'https://help.github.com/articles/resolving-a-merge' +
+          '-conflict-using-the-command-line/'
+      );
+      const params = {
+        repo: payloadData.payload.repository.name,
+        owner: payloadData.payload.repository.owner.login,
+        number: payloadData.payload.pull_request.number,
+        body:
+          'Hi @' +
+          payloadData.payload.pull_request.user.login +
+          '. Due to recent changes in the "develop" branch, ' +
+          'this PR now has a merge conflict. ' +
+          'Please follow this ' +
+          link +
+          ' if you need help resolving the conflict, ' +
+          'so that the PR can be merged. Thanks!',
+      };
+      expect(github.issues.createComment).toHaveBeenCalledWith(params);
+    });
+
+    it('adds merge conflict label', () => {
+      expect(github.issues.addLabels).toHaveBeenCalled();
+      const params = {
+        repo: payloadData.payload.repository.name,
+        owner: payloadData.payload.repository.owner.login,
+        number: payloadData.payload.pull_request.number,
+        labels: [mergeConflictLabel.name],
+      };
+      expect(github.issues.addLabels).toHaveBeenCalledWith(params);
+    });
+  });
+
+  describe('pull request merge does not result in merge conflict', () => {
+    beforeEach(async () => {
+      // Simulate a merged pull request.
+      payloadData.payload.pull_request.merged = true;
```

Comment:
Same here - how are two things being simulated for same PR?

Reply:
I'm creating a new pull request object via `const pullRequest = { ...payloadData.payload.pull_request };`
And the new pull request object is what is gotten as the response from the mock api

Reply:
Yes, that sounds good - just do a rename so that it is clear and add a comment as well.

Reply:
Just saw you added a rename. Instead of pullRequest2, give a meaningful name - eg: pullRequestToBeChecked.

Reply:
Done.